iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java

```diff
@@ -90,4 +93,16 @@ public StopSignalTree STOPSIGNAL() {
       )
     );
   }
+
+  public ExposeTree EXPOSE() {
+    return b.<ExposeTree>nonterminal(DockerLexicalGrammar.EXPOSE).is(
+      f.expose(b.token(DockerKeyword.EXPOSE), b.oneOrMore(PORT()))
+    );
+  }
+
+  public PortTree PORT() {
+    return b.<PortTree>nonterminal(DockerLexicalGrammar.PORT).is(
+      f.port(b.token(DockerLexicalGrammar.STRING_LITERAL))
```

Comment:
The logic of `<port>/<protocol>` should be handled in the grammar. WDYT? And what about ranges? They are not handled by the grammar or the factory.

Reply:
You are right, this is defeating the purpose of the grammar.